shared/javascripts/options.js

```diff
@@ -0,0 +1,387 @@
+    if (postingContentServerUrl !== undefined) {
+      try {
+        Privly.Options.setServerUrl(postingContentServerUrl);
+        ls.removeItem('posting_content_server_url');
+      } catch (ignore) {}
+    }
+    // Glyph
+    var glyphColor = ls.getItem('glyph_color');
+    var glyphCell = ls.getItem('glyph_cell');
+    if (glyphColor !== undefined && glyphCell !== undefined) {
+      try {
+        Privly.Options.setGlyph({
+          color: glyphColor,
+          cells: glyphCell.split(',').map(function (fill) {
+            return (fill === 'true');
+          })
+        });
+        ls.removeItem('glyph_color');
+        ls.removeItem('glyph_cell');
+      } catch (ignore) {}
+    }
+  };
+
+
+  // If this script is running as a background script
+  if (typeof chrome !== 'undefined' && chrome.runtime && chrome.runtime.getBackgroundPage) {
+    // Set event listeners to execute upgrade() function when
+    // the extension got installed.
+    chrome.runtime.onInstalled.addListener(function () {
+      Privly.Options.upgrade();
+    });
+
+    // Listen incoming messages to provide option interfaces
+    // for content scripts
+    chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
+      if (request.ask.indexOf('options/') === 0) {
+        var method = request.ask.split('/')[1];
+        if (Privly.Options[method] !== undefined) {
+          var returnValue = Privly.Options[method].apply(Privly.Options, request.params);
+          sendResponse(returnValue);
+        }
+      }
+    });
+  }
+
+
+  /**
+   * Whether Privly posting button is enabled
+   * @return {Boolean}
+   */
+  Privly.Options.isPrivlyButtonEnabled = function () {
+    var v = ls.getItem('options/privlyButton');
+    if (v === undefined) {
+      v = false;
+    }
+    return v;
+  };
+
+  /**
+   * Enable or disable the Privly posting button
+   * @param {Boolean} enabled
+   */
+  Privly.Options.setPrivlyButtonEnabled = function (enabled) {
+    if (typeof enabled !== 'boolean') {
+      throw new Error('invalid argument');
+    }
+    ls.setItem('options/privlyButton', enabled);
+    optionChanged('options/privlyButton', enabled);
+    return true;
+  };
+
+  /**
+   * Whether content injection is enabled
+   * @return {Boolean}
+   */
+  Privly.Options.isInjectionEnabled = function () {
+    var v = ls.getItem('options/injection');
+    if (v === undefined) {
+      v = true;
+    }
+    return v;
+  };
+
+  /**
+   * Enable or diable content injection
+   * @param {Boolean} enabled
+   */
+  Privly.Options.setInjectionEnabled = function (enabled) {
+    if (typeof enabled !== 'boolean') {
+      throw new Error('invalid argument');
+    }
+    ls.setItem('options/injection', enabled);
+    optionChanged('options/injection', enabled);
+    return true;
+  };
+
+  /**
+   * Validates a FQDN
+   * @param  {String}  domain Domain to validate
+   * @return {Boolean}
+   */
+  Privly.Options.isWhitelistDomainValid = function (domain) {
```

Comment:
This isn't checking that a domain is whitelisted, it is checking that the domain is valid. The function's name should be "isValidDomain".